webapp/tests/karma/unit/services/report-view-model-generator.js

```diff
@@ -0,0 +1,124 @@
+describe('ReportViewModelGenerator service', () => {
+
+  'use strict';
+
+  let service,
+      formatDataRecord,
+      lineageModelGenerator,
+      getSubjectSummaries,
+      getSummaries,
+      report,
+      formatted;
+
+  beforeEach(() => {
+    module('inboxApp');
+    module($provide => {
+      report = {
+        _id: 'my-report',
+        form: 'my-form',
+        fields: {
+          field1: 1,
+          field2: 2,
+          field3: 3
+        },
+        hidden_fields: ['field2'],
+        _attachments: {
+          content: { content_type: 'application/xml'},
+          'user-file/my-form/field1': { content_type: 'text/html' },
+          'user-file/my-form/field2': { something: '1' },
+          'user-file/my-form/fields/field21': { foo: 'bar' }
+        }
+      };
+
+      formatted = { formatted1: 1, formatted2: 2 };
+
+      lineageModelGenerator = { report: sinon.stub().resolves({ doc: report }) };
+      formatDataRecord = sinon.stub().resolves(formatted);
+      getSubjectSummaries = sinon.stub().resolves([{ summary: true, subject: 'subject' }]);
+      getSummaries = sinon.stub().resolves([{ summary: true }]);
+
+      $provide.value('LineageModelGenerator', lineageModelGenerator);
+      $provide.value('FormatDataRecord', formatDataRecord);
+      $provide.value('GetSubjectSummaries', getSubjectSummaries);
+      $provide.value('GetSummaries', getSummaries);
+      $provide.value('DB', {});
```

Comment:
Providing the values here is good, but I prefer to do the actual stubbing in each test. It makes it clearer to me what services are expected to be called during the test. If, for example, you stub something here which should never be called in one of the tests then it won't obviously fail if there is a bug in the code where it is called.